### PR TITLE
Av photogram industrial 2

### PR DESCRIPTION
**Pull request by AnnaVi11arrea1**

### CODE REVIEW

**Review by dgarcia13-99 — APPROVED** [review]
looks good! You keep everything nicely organized and indented. I especially like the layout of your routes page!

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
+class UsersController < ApplicationController
```
Comment (JoseMDO) [inline app/controllers/users_controller.rb]: Hi Anna, you actually do not need a users controller for these assignments. Devise takes care of all of this for us. It takes care of the routing, controllers, and views. Doing it manually can cause issues if it tries to interfere with the Devise controllers and views.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  has_many :leaders, through: :sent_follow_requests, source: :recepient
+
+  has_many :followers, through: :accepted_received_follow_requests, source: :sender
+
+  has_many :feed, through: :leaders, source: :own_photos
+
+  has_many :discover, through: :leaders, source: :liked_photos
+
+  # validates :username, presence: true, uniqueness: true
```
Comment (JoseMDO) [inline app/models/user.rb]: You can delete any commented out code if it is not being used.

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+  resources :likes
+  resources :comments
+  resources :followers
+  resources :photos
+  devise_for :users
+  devise_for :usernames
 
-  # Defines the root path route ("/")
   root "photos#index"
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
```
Comment (JoseMDO) [inline config/routes.rb]: You do not need any of these user routes. Devise takes care of these. Please delete these as well as your users controller

**Review by JoseMDO — CHANGES_REQUESTED** [review]
Please remove the Users controller and custom routes. Devise does all of the user functionality for us so this is excess code that is not required.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (AnnaVi11arrea1) [reply app/controllers/users_controller.rb]: fixed

File: config/routes.rb (hunk 3 above)
Comment (AnnaVi11arrea1) [reply config/routes.rb]: done

File: app/models/user.rb (hunk 2 above)
Comment (AnnaVi11arrea1) [reply app/models/user.rb]: comments removed from models and conrollers

**Review by JoseMDO — APPROVED** [review]
Thank you for correcting the users routes and controller. Great job!

**Pull request merged**